### ISSUE TITLE
MetaProperty nodes are not converted correctly

### ISSUE THREAD

**Issue by soda0289** [eslint/typescript-eslint-parser#194]
**What version of TypeScript are you using?**
2.2.1

**What version of `typescript-eslint-parser` are you using?**
2.0.0

**What code were you trying to parse?**

```ts
new foo.bar();
//OR
() => {
    new.target;
}
```

**What did you expect to happen?**
`new foo.bar()` should generate a NewExpression node
`new.target()` should generate a MetaProperty node

**What happened?**
They both generate invalid MetaProperty nodes

PR will be up soon
